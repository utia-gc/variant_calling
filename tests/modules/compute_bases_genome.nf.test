nextflow_process {

    name "Test Process compute_bases_genome"
    script "modules/compute_bases_genome.nf"
    process "compute_bases_genome"

    test("compute_bases_genome computes the number of bases in a reference genome") {
        // instantiate helper objects
        def refs = new RefsEnsembl()

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            process {
                """
                input[0] = file("${refs.getGenomeIndex()}")
                """
            }
        }

        then {
            // test process success
            assert process.success

            // test expected number bases
            with(process.out.bases) {
                assert Long.valueOf(get(0)) == 230218L
            }

            // test reproducibility
            assert snapshot(process.out).match()
        }

    }

    test("compute_bases_genome computes the number of bases in a large reference genome and outputs a number") {
        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            process {
                """
                input[0] = file("https://github.com/utia-gc/ngs-test/raw/b34247fcd38695c8a07abdb8fa521424920a97db/data/references/GCF_002263795.3_ARS-UCD2.0_genomic.fna.fai")
                """
            }
        }

        then {
            // test process success
            assert process.success

            // test expected number bases
            with(process.out.bases) {
                assert Long.valueOf(get(0)) > 2_000_000_000L
            }

            // test reproducibility
            assert snapshot(process.out).match()
        }

    }

}
