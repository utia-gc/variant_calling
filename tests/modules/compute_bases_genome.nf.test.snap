{
    "compute_bases_genome computes the number of bases in a reference genome": {
        "content": [
            {
                "0": [
                    "230218"
                ],
                "bases": [
                    "230218"
                ]
            }
        ],
        "timestamp": "2024-01-19T11:28:19.77708076"
    },
    "compute_bases_genome computes the number of bases in a large reference genome and outputs a number": {
        "content": [
            {
                "0": [
                    "2770686120"
                ],
                "bases": [
                    "2770686120"
                ]
            }
        ],
        "timestamp": "2024-01-26T10:02:40.34260478"
    }
}